### PR TITLE
analyzer: Port package managers to use ServiceLoader

### PR DESCRIPTION
**Pull request by sschuberth**
Also align other minor code aspects with the Scanner classess which
already use ServiceLoader.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/791)

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/NPM.kt [hunk 1]
```diff
@@ -270,7 +274,7 @@ open class NPM(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConf
 
             val module = Package(
                     id = Identifier(
-                            provider = NPM.toString(),
+                            provider = "NPM",
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/NPM.kt]: I see why you cannot use `toString` anymore, but I don't like the hardcoded provider Strings. But I also have no good idea how to fix it, e.g. a `name` property in `PackageManager` would work here, but not when `Gradle` needs `"Maven"` for the package references.

File: analyzer/src/test/kotlin/PackageManagerTest.kt [hunk 2]
```diff
@@ -39,25 +39,26 @@ class PackageManagerTest : WordSpec({
...
-            result[GoDep] shouldBe listOf(File(projectDir, "Gopkg.toml"))
-            result[Gradle] shouldBe listOf(File(projectDir, "build.gradle"))
-            result[Maven] shouldBe listOf(File(projectDir, "pom.xml"))
-            result[NPM] shouldBe listOf(File(projectDir, "package.json"))
-            result[PhpComposer] shouldBe listOf(File(projectDir, "composer.json"))
-            result[PIP] shouldBe listOf(File(projectDir, "setup.py"))
-            result[SBT] shouldBe listOf(File(projectDir, "build.sbt"))
-            result[Stack] shouldBe listOf(File(projectDir, "stack.yaml"))
-            result[Yarn] shouldBe listOf(File(projectDir, "yarn.lock"))
+            result[Bundler.Factory()] shouldBe listOf(File(projectDir, "Gemfile"))
```
Comment (mnonnenmacher) [inline analyzer/src/test/kotlin/PackageManagerTest.kt]: There are no `hashCode` and `equals` implementations in PackageManagerFactory, so this is bound to fail, right?

Comment (sschuberth) [reply analyzer/src/test/kotlin/PackageManagerTest.kt]: Right, on it.

File: analyzer/src/main/kotlin/managers/NPM.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/NPM.kt]: I had similar thoughts which is why I created the `pm-tostring` branch in parallel which has https://github.com/heremaps/oss-review-toolkit/commit/8e15035f233745ad2b5b29475c2919c58fef81ab. Let's discuss about that separately.

**Pull request merged**